mentat/embeddings.py

```diff
@@ -21,36 +23,52 @@
 
 EMBEDDING_MODEL = "text-embedding-ada-002"
 EMBEDDING_DIM = 1536
+MAX_SIMULTANEOUS_REQUESTS = 10
 
 
 class EmbeddingsDatabase:
-    # { sha256 : [ EMBEDDING_DIM floats ] }
-    _dict: dict[str, list[float]] = dict[str, list[float]]()
-
     def __init__(self, output_dir: Path | None = None):
-        if output_dir is None:
-            output_dir = mentat_dir_path
-        os.makedirs(output_dir, exist_ok=True)
-        self.path = Path(output_dir) / "embeddings.json.gz"
-        if self.path.exists():
-            try:
-                with gzip.open(self.path, "rt") as f:
-                    self._dict = json.load(f)
-            except gzip.BadGzipFile:
-                logging.warning(f"Could not load embeddings from {self.path}.")
+        self.output_dir = output_dir or mentat_dir_path
+        os.makedirs(self.output_dir, exist_ok=True)
+        self.path = Path(self.output_dir) / "embeddings.sqlite3"
+        self._connect()
+
+    def _connect(self):
+        self.conn = sqlite3.connect(self.path)
+        with self.conn:
+            self.conn.execute(
+                "CREATE TABLE IF NOT EXISTS embeddings "
+                "(checksum TEXT PRIMARY KEY, vector BLOB)"
+            )
 
-    def save(self):
-        with gzip.open(self.path, "wt") as f:
-            json.dump(self._dict, f)
+    def set(self, items: dict[str, list[float]]):
+        with self.conn:
+            self.conn.executemany(
```

Comment:
I think here and line 38 you should write `with self.conn as db` and then `db.executemany` instead of repeating `self.conn`